src/transformers/models/whisper/generation_whisper.py

```diff
@@ -474,11 +473,8 @@ def generate(
                 "The input name `inputs` is deprecated. Please make sure to use `input_features` instead.",
                 FutureWarning,
             )
-        # 1. copy generation config
-        if generation_config is None:
-            generation_config = copy.deepcopy(self.generation_config)
-        else:
-            generation_config = copy.deepcopy(generation_config)
+        # 1. prepare generation config
+        generation_config, kwargs = self._prepare_generation_config(generation_config, **kwargs)
```

Comment:
This function from the main `generate` body (`_prepare_generation_config`) pulls generation parameterization from `kwargs` into `generation_config`. 

Some Whisper-based doctests were incorrect without this functionality.